### PR TITLE
feat(text field): add new component [WIP]

### PR DESCRIPTION
**Pull request by yasminnvaz**
# Description

Create new component

## Type of change

- [x] Feat: A new feature (non-breaking change which adds functionality)
- [ ] Refactor: A code change that neither fixes a bug nor adds a feature
- [ ] Breaking change: fix or feature that would cause existing functionality to not work as expected
- [ ] Docs: Documentation only changes

# How Has This Been Tested?

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings or errors;
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] My commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) Guidelines;

### CODE REVIEW

File: packages/react/src/components/Input/Input.props.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
+import { IconName } from '@naturacosmeticos/natds-icons'
+import { Size } from '@naturacosmeticos/natds-themes'
+
+export type InputSize = keyof Pick<Size, 'medium' | 'mediumX'>
+export type InputType = 'text' | 'password' | 'multiline'
+export type Feedback = 'success' | 'error'
+
+export interface BaseProps {
```
Comment (lauralucca) [inline packages/react/src/components/Input/Input.props.ts:8]: Seria bom adicionar o `aria-labelledby` e o `aria-label` no input e na label. Inclusive reforçar a regra de usar um ou o outro é mais uma oportunidade de exercitar aquilo que aprendemos sobre TS

File: packages/react/src/components/Input/Input.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+              placeholder={placeholder}
+              readOnly={readOnly}
+              rows={minRows}
+              id={id}
+            />
+          )
+          : (
+            <input
+              className={classes.input}
+              data-testid="ds-input"
```
Comment (lauralucca) [inline packages/react/src/components/Input/Input.tsx:49]: Mesma coisa que disse sobre o `testid` padrão acima.

File: packages/react/src/components/Input/InputLabel/InputLabel.styles.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+const styles = createUseStyles((theme: Theme) => ({
+  label: {
+    color: getFeedbackTextColor(theme),
+    display: ({ label }: InputLabelProps) => (label ? 'inline-block' : 'none'),
+    fontFamily: [theme.typography.fontFamily.primary, theme.typography.fontFamily.secondary],
+    fontSize: 14,
+    letterSpacing: 0.11,
+    marginBottom: theme.spacing.micro
```
Comment (lauralucca) [inline packages/react/src/components/Input/InputLabel/InputLabel.styles.ts:25]: Pelo que vi na ui doc a Label devia usar a tipografia `subtitle2`

File: packages/react/src/components/Input/InputHelperText/InputHelperText.styles.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+const styles = createUseStyles((theme: Theme) => ({
+  helperText: {
+    alignItems: 'center',
+    color: getFeedbackTextColor(theme),
+    display: 'flex',
+    fontFamily: [theme.typography.fontFamily.primary, theme.typography.fontFamily.secondary],
+    fontSize: 12,
+    letterSpacing: 0.4,
+    marginTop: theme.spacing.micro,
```
Comment (lauralucca) [inline packages/react/src/components/Input/InputHelperText/InputHelperText.styles.ts:27]: Pelo que vi na ui doc a HelerText devia usar a tipografia `caption`

File: packages/react/src/components/TextField/TextField.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+export const ActionImage: Story<TextFieldProps> = (args) => (
+  <StoryContainer>
+    <TextField {...args} />
+  </StoryContainer>
+)
+
+ActionImage.args = {
+  ...Playground.args,
+  action: 'image',
+  src: 'https://cdn.jsdelivr.net/npm/@naturacosmeticos/natds-react@2.0.0-alpha.DSY-1963.53.0/dist/assets/mastercard@2x.png',
```
Comment (lauralucca) [inline packages/react/src/components/TextField/TextField.stories.tsx:140]: Aqui é bom trocar pela imagem padrão que o Lucas pediu quando foi feita a primeira revisão da doc.

**Review by lauralucca — CHANGES_REQUESTED** [review]

File: packages/react/src/components/TextField/TextField.test.tsx [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+  id: 'example_to_test'
+}
+
+describe('TextField component', () => {
+  it('should render correctly with default props', () => {
+    const { styles, component } = renderWithTheme(<TextField {...defaultProps} />)
+
+    expect([styles.toString(), component.container]).toMatchSnapshot()
+  })
+  it('should render correctly have a label', () => {
```
Comment (arielwb) [inline packages/react/src/components/TextField/TextField.test.tsx:21]: should render correctly when have a label

File: packages/react/src/components/TextField/TextField.test.tsx [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+    const { styles, component } = renderWithTheme(<TextField {...defaultProps} size="medium" />)
+
+    expect([styles.toString(), component.container]).toMatchSnapshot()
+  })
+  it('should render correctly when it has an error feedback', () => {
+    const { styles, component } = renderWithTheme(<TextField {...defaultProps} feedback="error" />)
+
+    expect([styles.toString(), component.container]).toMatchSnapshot()
+  })
+  it('should render correctly when it has an success feedback', () => {
```
Comment (arielwb) [inline packages/react/src/components/TextField/TextField.test.tsx:36]: should render correctly when it has A success feedback
sem o n :P

**Review by arielwb — APPROVED** [review]

**Review by lauralucca — APPROVED** [review]
LGTM

**Pull request merged**